stdlib/arg.ml

```diff
@@ -233,10 +239,19 @@ let parse_argv_dynamic ?(current=current) argv speclist anonfun errmsg =
         | Tuple specs ->
             List.iter treat_action specs;
         | Rest f ->
-            while !current < l - 1 do
-              f argv.(!current + 1);
+            while !current < (Array.length !argv) - 1 do
+              f !argv.(!current + 1);
               consume_arg ();
             done;
+        | Expand f ->
+            if not expand then
+              raise (Stop (Message ("Expand is not allowed")));
```

Comment:
This is in fact a misuse of the API so it should raise `Invalid_argument`. I would put some informative message to help the programmer debug the problem: `"Arg.Expand is only allowed with Arg.parse_and_expand_argv_dynamic"`